yb-voyager/src/migassessment/assessmentDB.go

```
@@ -385,3 +397,151 @@ func (adb *AssessmentDB) Query(query string, args ...interface{}) (*sql.Rows, er
+		return fmt.Errorf("failed to parse PGSS CSV: %w", err)
+	}
+
+	if len(entries) == 0 {
+		log.Warnf("No valid PGSS entries found in %s", filePath)
+		return nil
+	}
+
+	log.Infof("inserting PGSS entries into %s table", DB_QUERIES_SUMMARY)
+	successCount, err := adb.InsertPgssEntries(entries)
+	if err != nil {
+		return fmt.Errorf("failed to insert PGSS entries: %w", err)
+	}
+
+	log.Infof("Successfully loaded %d PGSS entries into %s table", successCount, DB_QUERIES_SUMMARY)
+	return nil
+}
+
+func (adb *AssessmentDB) InsertPgssEntries(entries []pgss.QueryStats) (int, error) {
+	if len(entries) == 0 {
+		return 0, nil
+	}
+
+	// Prepare statement for batch insertion
+	insertSQL := `INSERT INTO db_queries_summary 
+		(queryid, query, calls, rows, total_exec_time, mean_exec_time, min_exec_time, max_exec_time, stddev_exec_time) 
+		VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)
+		ON CONFLICT(queryid) DO UPDATE SET
+			query = excluded.query,
+			calls = excluded.calls,
+			rows = excluded.rows,
+			total_exec_time = excluded.total_exec_time,
+			mean_exec_time = excluded.mean_exec_time,
+			min_exec_time = excluded.min_exec_time,
+			max_exec_time = excluded.max_exec_time,
+			stddev_exec_time = excluded.stddev_exec_time`
+
+	stmt, err := adb.db.Prepare(insertSQL)
+	if err != nil {
+		return 0, fmt.Errorf("failed to prepare PGSS insert statement: %w", err)
+	}
+	defer stmt.Close()
+
+	successCount := 0
+	errorCount := 0
+
+	// Insert each entry
+	for _, entry := range entries {
+		_, err = stmt.Exec(
+			entry.QueryID,
+			entry.Query,
+			entry.Calls,
+			entry.Rows,
+			entry.TotalExecTime,
+			entry.MeanExecTime,
+			entry.MinExecTime,
+			entry.MaxExecTime,
+			entry.StddevExecTime,
+		)
+		if err != nil {
+			log.Warnf("Failed to insert PGSS entry for queryid %d: %v", entry.QueryID, err)
+			errorCount++
+			continue
+		}
+		successCount++
+	}
+
+	if errorCount > 0 {
+		log.Warnf("PGSS insertion completed with %d errors out of %d total entries", errorCount, len(entries))
```

why just warn? shouldn't we return the error properly, and fail?